extensions/go/README.md

```diff
@@ -2,13 +2,65 @@
+## Usage
+
+1. Enable the `sourcegraph/go` extension:
+    - On Sourcegraph.com, visit the [extension page](https://sourcegraph.com/extensions/sourcegraph/go) to enable it.
+    - On a self-hosted Sourcegraph instance, select **User menu > Extensions**, search for `sourcegraph/go`, and enable it.
+1. Visit any Go code file on Sourcegraph.
+1. Hover over a token in the code file.
+    - See a description of the token.
+    - Click **Go to definition** (if available) to go to the token's definition.
+    - Click **Find references** to see all references to the token.
+
+### On your code host
+
+This extension adds the same features to code files and diffs on your code host if you're using the [Sourcegraph browser extension](https://docs.sourcegraph.com/integration/browser_extension). To use it on your code host:
+
+1. Follow the [usage steps](#usage) above to enable this extension.
+1. Install the [Sourcegraph browser extension](https://docs.sourcegraph.com/integration/browser_extension).
+    - If you're using it with a self-hosted Sourcegraph instance, enter the Sourcegraph instance URL into the Sourcegraph browser extension options menu. Then click the gear icon and enable _Experimental features: Use extensions_.
+1. Visit any file on your code host and hover over a token to see a description of the token, a **Go to definition** action (if available), and a **Find references** action.
+
+## Basic code intelligence
+
+This extension comes with built-in code intelligence provided by [search-based heuristics](https://docs.sourcegraph.com/user/code_intelligence/basic_code_intelligence). Because this extension uses text-based heuristics, its definition and reference results may not be precise:
+
+-   "Go to definition" on a token goes to the definition found by [universal-ctags](https://github.com/universal-ctags/ctags), a cross-language parsing suite.
+-   "Find references" on a token finds all instances of token (with the same case) in the current repository and other repositories.
+
+These heuristics work well for tokens with unique names, such as `render_to_view` or `TLSConfig`. They do not work well for ambiguous tokens, such as `open` or `file`.
+
+### Large repositories
+
+Heuristic code intelligence will perform a search query in the commit you are viewing. This may cause performance issues if the commit is not indexed and the repository is large. After a timeout period with no results, an index-only search will be performed. This type of query may return results for a commit other than the one you are currently viewing. The default timeout period is five seconds, but can be lowered by adding the following to your Sourcegraph global settings:
+
+    ```json
+    "basicCodeIntel.unindexedSearchTimeout": 1000
+    ````
+
+For organizations that organize code in a monorepo, it may never be useful to perform an un-indexed search. Index-only search queries can be forced by adding the following to your Sourcgraph global settings.
```

Comment:
```suggestion
For organizations that organize code in a monorepo, it may never be useful to perform an un-indexed search. To force only indexed search queries, add the following to your Sourcgraph global settings.
```

Inspired by https://about.sourcegraph.com/handbook/communication/style_guide#avoid-jargon-and-acronyms, https://about.sourcegraph.com/handbook/communication/style_guide#clarity

Reply:
Updated in all three places.